google-cloud-compute/src/main/java/com/google/cloud/compute/testing/ManagedResourceHelper.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.compute.testing;
+
+import com.google.cloud.compute.Address;
+import com.google.cloud.compute.AddressId;
+import com.google.cloud.compute.Compute;
+import com.google.cloud.compute.Disk;
+import com.google.cloud.compute.DiskId;
+import com.google.cloud.compute.Image;
+import com.google.cloud.compute.Instance;
+import com.google.cloud.compute.Network;
+import com.google.cloud.compute.NetworkId;
+import com.google.cloud.compute.Operation;
+import com.google.cloud.compute.Snapshot;
+import com.google.cloud.compute.SnapshotId;
+import com.google.cloud.compute.Subnetwork;
+import com.google.cloud.compute.SubnetworkId;
+import com.google.common.base.Function;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ManagedResourceHelper {
+  private final Compute compute;
+  private final List<ManagedResource<?>> resources;
+
+  private ManagedResourceHelper(Compute compute) {
+    this.compute = compute;
+    this.resources = new ArrayList<>();
+  }
+
+  public static ManagedResourceHelper create(Compute compute) {
+    return new ManagedResourceHelper(compute);
+  }
+
+  private class ManagedResource<T> {
+    final T res;
+    Function<T, Void> func;
+
+    private ManagedResource(T res) {
+      this.res = res;
+
+      @SuppressWarnings("unchecked")
+      Function<T, Void> localFunc = (Function<T, Void>) getDeleteMethod(res);
+
+      this.func = localFunc;
```

Comment:
yeah, this is ugly. The reason I'd like to have a seemingly repeated statement is because then I can limit the scope of `@SuppressWarnings("unchecked")`